benchmark/solver/solver.cpp

```diff
@@ -486,32 +490,49 @@ int main(int argc, char *argv[])
             std::clog << "Running test case: " << test_case << std::endl;
             std::ifstream mtx_fd(test_case["filename"].GetString());
 
-            using Vec = gko::matrix::Dense<>;
+            using Vec = gko::matrix::Dense<etype>;
             std::shared_ptr<gko::LinOp> system_matrix;
             std::unique_ptr<Vec> b;
             std::unique_ptr<Vec> x;
             if (FLAGS_overhead) {
                 system_matrix = gko::initialize<Vec>({1.0}, exec);
-                b = gko::initialize<Vec>({std::nan("")}, exec);
+                b = gko::initialize<Vec>(
+                    {std::numeric_limits<rc_etype>::quiet_NaN()}, exec);
                 x = gko::initialize<Vec>({0.0}, exec);
             } else {
                 auto data = gko::read_raw<etype>(mtx_fd);
                 system_matrix = share(formats::matrix_factory.at(
                     test_case["optimal"]["spmv"].GetString())(exec, data));
+                auto vec_size =
+                    gko::dim<2>{system_matrix->get_size()[0], FLAGS_nrhs};
                 if (test_case.HasMember("rhs")) {
                     std::ifstream rhs_fd{test_case["rhs"].GetString()};
                     b = gko::read<Vec>(rhs_fd, exec);
                 } else {
-                    b = create_matrix<etype>(
-                        exec,
-                        gko::dim<2>{system_matrix->get_size()[0], FLAGS_nrhs},
-                        engine, FLAGS_random_rhs);
+                    // TODO: Remove (only present to perform benchmark for
```

Comment:
Is it possible to resolve this todo now? Seems like you're making the sin RHS available permanently, in which case you can remove this comment.

Reply:
I still want to remove it. The sin-benchmark was added as the last commit, which I will simply remove before actually merging it (this is the reason this PR is currently marked as `WIP`).
I plan to keep all other changes of this PR, just the sinus benchmark portion will be removed.

Reply:
Oh sorry, I completely missed the label change. I should have waited a bit, but it should be fine I guess.